R/plot_taxa.R

```diff
@@ -422,3 +758,548 @@ format_for_comm_plots <- function(observation, id) {
     dswide = dswide)
   return(res)
 }
+
+
+
+
+
+
+
+
+#' Plot taxa ranks
+#'
+#' @param dataset (list) An ecocomDP formatted dataset
+#' @param flat_data (tbl_df, tbl, data.frame) A flat data.frame created from an \code{ecocomDP} formatted dataset using \code{ecocomDP::flatten_data}.
+#' @param observation (tbl_df, tbl, data.frame) The observation table.
+#' @param taxon (tbl_df, tbl, data.frame) The taxon table.
+#' @param id (character) Identifier of dataset to be used in plot subtitles.
+#' @param facet_var (character) Name of column to use for faceting.
+#' @param facet_scales (character) Should scales be free ("free", default value), fixed ("fixed"), or free in one dimension ("free_x", "free_y")?
+#' @param alpha (numeric) Alpha-transparency scale of data points. Useful when many data points overlap. Allowed values are between 0 and 1, where 1 is 100\% opaque. Default is 1.
+#' 
+#' @return (gg, ggplot) A gg, ggplot object if assigned to a variable, otherwise a plot to your active graphics device
+#' 
+#' @import dplyr
+#' @import ggplot2
+#' @import tidyr
+#' 
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' # plot ecocomDP formatted dataset
+#' plot_taxa_rank(
+#'   dataset = ants_L1)
+#' 
+#' plot_taxa_rank(
+#'   dataset = ants_L1,
+#'   facet_var = "location_id")
+#' 
+#' # plot flattened ecocomDP dataset
+#' plot_taxa_rank(
+#'   flat_data = flatten_dataset(ants_L1),
+#'   facet_var = "location_id")
+#' 
+#' # plot ecocomDP observation table
+#' plot_taxa_rank(
+#'   observation = ants_L1[[1]]$tables$observation,
+#'   taxon = ants_L1[[1]]$tables$taxon,
+#'   facet_var = "location_id")
+#' 
+#' # tidy syntax, filter data by date
+#' ants_L1 %>% 
+#'   flatten_dataset() %>% 
+#'   dplyr::filter(
+#'     lubridate::as_date(datetime) > "2003-07-01") %>%
+#'   plot_taxa_rank(flat_data = .)
+#' 
+#' # tidy syntax, filter data by site ID
+#' ants_L1 %>% 
+#'   flatten_dataset() %>% 
+#'   dplyr::filter(
+#'     as.numeric(location_id) > 4) %>%
+#'   plot_taxa_rank(flat_data = .)
+#' }
+plot_taxa_rank <- function(
```

Comment:
Same comments as `plot_taxa_accum_sites()`

Reply:
that approach doesn't work for this function because it uses multiple tables -- the taxon and observation tables.

Reply:
Use the argument subsuming approach offered in `plot_taxa_accum_sites()`?

Reply:
@sokole Add deprecation notice and handling for observation param.